src/sentry/runner/commands/notifications.py

```diff
@@ -120,3 +186,50 @@ def list_cmd() -> None:
         click.echo(f"• category: {category}")
         for source in sources:
             click.echo(f"  ◦ source: {source['source']}, class: {source['class']}")
+
+
+@click.option(
+    "-p", "--provider", help="the integration provider e.g. slack, discord, msteams", required=True
+)
+@click.option("-o", "--organization_slug", help="Organization slug", required=True)
+@notifications.command("list-integrations")
+def list_integrations(organization_slug: str | None, provider: str) -> None:
+    """
+    List all integrations available for a given provider
+    - Optionally can be given an organization slug to show only integrations for that org
+    """
+    from sentry.runner import configure
+
+    configure()
+
+    from sentry.constants import ObjectStatus
+    from sentry.integrations.models.organization_integration import OrganizationIntegration
+    from sentry.integrations.types import IntegrationProviderSlug
+    from sentry.models.organizationmapping import OrganizationMapping
+
+    try:
+        provider = IntegrationProviderSlug(provider)
+    except ValueError:
+        click.echo(f"Invalid provider: {provider}")
+        return
+
+    try:
+        organization = OrganizationMapping.objects.get(slug=organization_slug)
+    except OrganizationMapping.DoesNotExist:
+        click.echo(f"Organization {organization_slug} not found!")
+        return
+
+    # Get organization integrations that belong to this organization and match our provider
+    organization_integrations = OrganizationIntegration.objects.filter(
+        integration__provider=provider,
+        integration__status=ObjectStatus.ACTIVE,
+        organization_id=organization.organization_id,
```

Comment:
This line confused me, but the `organization` variable is actually just a `mapping`, so I guess this a nit for naming but doesn't really matter.